docs/check/overview.md

```diff
@@ -0,0 +1,583 @@
+# Check
+The [Check API](https://openfga.dev/api/service#/Relationship%20Queries/Check) looks up if a particular user/subject has specific relationship with a given object. 
+
+It is a forward expansion algorithm that starts at a given `object#relation` and iteratively and recursively expands relationships until a particular user/subject is found or until all paths of the expansion have been visited and no such paths were found. It can be viewed as a tree traversal or a directed graph traversal. 
+
+As relationships are expanded the Check algorithm follows relationship rewrite rules, and these rewrite rules define one or more paths that must be evaluated. For example, given the following FGA model:
+```
+model
+  schema 1.1
+
+type user
+
+type folder
+ relations
+   define viewer: [user]
+
+type document
+  relations
+    define parent: [folder]
+    define editor: [user] or editor from parent
+    define viewer: editor
+```
+The `document#viewer` relation is "rewritten" indirectly as `document#editor`, which means that in order to evaluate a specific relationship of the form `Check(document:1#viewer@user:jon)` we must actually rewrite that lookup as `Check(document:1#editor@user:jon)`. Similarly, the `document#editor` relationship is rewritten using a set union (e.g. `or` keyword) such that if a user is either directly an editor of the document or an editor on the parent folder, then the user can edit the document. This kind of rewritten behavior allows us to model hierarchical semantics. These "rewrites" make up the base-line behavior behind the semantics of the FGA graph traversal, and we explain each of them with examples in further detail below.
+
+## Definitions
+Familiarity with basic [OpenFGA Concepts](https://openfga.dev/docs/concepts) is expected, but the following are some terms used throughout this document that are more specific to the content herein:
+
+* **Relation Rewrite (Rewrite)** - a rule that expresses an FGA relationship as a composition of other relationships. For example, you can "rewrite" a viewer relationship as editor or express a hierarchy between two objects. These rules can also be composed using set operations including union, intersection, and exclusion.
```

Comment:
what do you think about including a drawing showing all the possible rewrites? (add union, intersection and exclusion)

![image](https://github.com/openfga/openfga/assets/5374887/27c34c45-d1ba-4413-92cb-4fc552ceb426)

Reply:
I translated this to a diagram (including set rewrites) and have now included it 👍